applications/vanilla/modules/class.newdiscussionmodule.php

```diff
@@ -123,7 +125,10 @@ public function toString() {
                 $url = $this->GuestUrl.'?Target='.$url;
             }
 
-            $this->addButton(t(val('AddText', $type)), $url);
+            // Check whether to display in dropdown or as a separate button.
+            $asOwnButton = $buttonsConfig[$type['Singular']]['AsOwnButton'] ?? false;
+
+            $this->addButton(t(val('AddText', $type)), $url, $asOwnButton);
```

Comment:
The `val()` function is now deprecated. Can you change to `$type['AddText'] ?? ''`?